### PR TITLE
Improve proxy support for EventHubClient and AuthClient

### PR DESCRIPTION
**Pull request by RAVEENSR**
## Purpose
Improve proxy support for EventHubClient and AuthClient

## Goals
Improve proxy support for EventHubClient and AuthClient

## Automation tests
 - Unit tests 
   > Code coverage information
 - Integration tests
   > Details about the test cases and coverage

## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java [hunk 1]
```diff
@@ -18,127 +18,130 @@
...
-    private static final String KEYSTORE_TYPE = "JKS";
-    private static final String PROTOCOL = "TLS";
-    private static final String ALGORITHM = "SunX509";
 
-    public static okhttp3.OkHttpClient getProxyClient(Map<String, String> properties) {
+    public static Client getClient(Map<String, String> properties) {
+        return getFeignHttpClient(properties, HTTPS_PROTOCOL);
+    }
 
+    public static ApacheHttpClient getFeignHttpClient(Map<String, String> properties, String protocol) {
```
Comment (ruks) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java:71]: should be private

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java [hunk 2]
```diff
@@ -18,127 +18,130 @@
...
+            log.error("Error while getting http client connection manager", e);
         }
-    }
-
-    private static SSLSocketFactory initSSLConnection(KeyStore keyStore, String keyStorePassword) throws
-            NoSuchAlgorithmException, UnrecoverableKeyException, KeyStoreException, KeyManagementException {
 
-        KeyManagerFactory keyManagerFactory = KeyManagerFactory.getInstance(ALGORITHM);
-        keyManagerFactory.init(keyStore, keyStorePassword.toCharArray());
+        HttpHost host = new HttpHost(proxyHost, proxyPort, protocol);
```
Comment (ruks) [inline component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java:89]: cant we just use the proxyProtocol here?

Comment (RAVEENSR) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java:89]: You mean create http host as `new HttpHost(protocol);`?

Comment (RAVEENSR) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java]: fixed

File: component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java (hunk 1 above)
Comment (RAVEENSR) [reply component/publisher-client/src/main/java/org/wso2/am/analytics/publisher/auth/AuthProxyUtils.java]: fixed

**Review by ruks — APPROVED** [review]

**Pull request merged**